plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/GitHook.kt

```diff
@@ -0,0 +1,155 @@
+package org.jlleitschuh.gradle.ktlint
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.file.ProjectLayout
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.intellij.lang.annotations.Language
+import java.io.File
+import javax.inject.Inject
+
+internal const val FILTER_INCLUDE_PROPERTY_NAME = "internalKtlintGitFilter"
+
+@Language("Bash")
+internal val shShebang = """
+#!/bin/sh
```

Comment:
@Tapchicoma I had to change this to `#!/bin/bash` on my machine, which is Elementary OS 5 / Ubuntu 18.04 - do you think it makes sense to change this to bash?

Reply:
Why do you need this change? :thinking:

Reply:
I'm guessing it's because Elementary  OS doesn't ship with `sh`, only `bash`.

Reply:
Seems so: https://askubuntu.com/questions/537669/script-doesnt-work-when-it-is-run-using-bin-sh

I will check what bashisms this script is using.

Reply:
```
$ checkbashisms .git/hooks/pre-commit
possible bashism in .git/hooks/pre-commit line 7 (alternative test command ([[ foo ]] should be [ foo ])):
if [[ -z "$CHANGED_FILES" ]]; then
possible bashism in .git/hooks/pre-commit line 20 (alternative test command ([[ foo ]] should be [ foo ])):
    if [[ -f $file ]]; then
possible bashism in .git/hooks/pre-commit line 23 (<<< here string):
done <<< "$CHANGED_FILES"
```

Reply:
I do actually have a `/bin/sh` on my disk but I guess it is some less advanced shell. And yes it was failing on line 23 `done <<< "$CHANGED_FILES"`

Reply:
Yeah my `sh` corresponds to `dash`.

Reply:
I will try to fix this in a separate PR. Thank you for letting us know it :)